pkg/auth/service.go

```diff
@@ -1264,7 +1264,7 @@ func (a *APIAuthService) GetUserByID(ctx context.Context, userID string) (*model
 	if err != nil {
 		return nil, fmt.Errorf("userID as int64: %w", err)
 	}
-	return a.getFirstUser(ctx, userKey{id: userID}, &ListUsersParams{Id: &intID})
+	return a.getFirstUser(ctx, userKey{id: userID}, &ListUsersParams{Id: &intID, Amount: paginationAmount(1)})
```

Comment:
currently, the getFirstUser method returns an error if more than one user meets the filter, by limiting it to 1 we lose that. If you believe this check isn't needed then please remove validation in getFirstUser.
Also thinking maybe it will be better to add the Amount inside the getFirstUser method

Reply:
Right, very bad call to handle this one in this level - my bad.
I'll handle the one limit inside the getFirstUser and by check `hasMore` instead of length of the results.